### PR TITLE
[10.0][IMP] web_switch_context_warning

### PR DESCRIPTION
**Pull request by AdriaGForgeFlow**
Previous version would not show the warning with the module web_responsive installed nor with the enterprise version of Odoo, with this PR it works now.

Since now the module allows detection of user, company and database switch, the module has been renamed to web_switch_context_warning.

cc ~ @Eficent

### CODE REVIEW

**Review by JordiBForgeFlow — APPROVED** [review]
:+1: Tested in runbot by switching companies, using web responsive. Works as expected.

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM

File: web_switch_company_warning/__manifest__.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
     "name": "Multicompany - Switch Company Warning",
     "summary": "Show a warning if current company has been switched"
                " in another tab or window.",
-    "version": "10.0.0.1.1",
+    "version": "10.0.0.1.2",
     "category": "web",
     "website": "http://akretion.com",
```
Comment (tbaden) [inline web_switch_company_warning/__manifest__.py]: could you please change the website to https://github.com/OCA/web/ as well in this PR

**Comment by tbaden** [comment]
could you please fix there travis warnings:
<details><summary>logs</summary>

> ************* Module web_switch_company_warning
> web_switch_company_warning/static/src/js/switch_company_warning.js:1: [W7903(javascript-lint), ] Use the function form of 'use strict'. [Error/strict]
> web_switch_company_warning/static/src/js/switch_company_warning.js:3: [W7903(javascript-lint), ] Use the function form of 'use strict'. [Error/strict]
> web_switch_company_warning/static/src/js/switch_company_warning.js:6: [W7903(javascript-lint), ] Expected space or tab after '//' in comment. [Error/spaced-comment]
> web_switch_company_warning/static/src/js/switch_company_warning.js:7: [W7903(javascript-lint), ] Expected space or tab after '//' in comment. [Error/spaced-comment]
> web_switch_company_warning/static/src/js/switch_company_warning.js:10: [W7903(javascript-lint), ] Comments should not begin with a lowercase character. [Error/capitalized-comments]
> web_switch_company_warning/static/src/js/switch_company_warning.js:10: [W7903(javascript-lint), ] Expected space or tab after '//' in comment. [Error/spaced-comment]
> web_switch_company_warning/static/src/js/switch_company_warning.js:15: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
> web_switch_company_warning/static/src/js/switch_company_warning.js:18: [W7903(javascript-lint), ] Line 18 exceeds the maximum line length of 80. [Error/max-len]
> web_switch_company_warning/static/src/js/switch_company_warning.js:23: [W7903(javascript-lint), ] Expected space(s) after "if". [Error/keyword-spacing]
> web_switch_company_warning/static/src/js/switch_company_warning.js:32: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
> web_switch_company_warning/static/src/js/switch_company_warning.js:34: [W7903(javascript-lint), ] Missing trailing comma. [Error/comma-dangle]
> web_switch_company_warning/static/src/js/switch_company_warning.js:38: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
> web_switch_company_warning/static/src/js/switch_company_warning.js:41: [W7903(javascript-lint), ] Line 41 exceeds the maximum line length of 80. [Error/max-len]
> web_switch_company_warning/static/src/js/switch_company_warning.js:47: [W7903(javascript-lint), ] Missing trailing comma. [Error/comma-dangle]
> web_switch_company_warning/static/src/js/switch_company_warning_worker.js:1: [W7903(javascript-lint), ] Use the function form of 'use strict'. [Error/strict]
> web_switch_company_warning/static/src/js/switch_company_warning_worker.js:2: [W7903(javascript-lint), ] Expected space or tab after '//' in comment. [Error/spaced-comment]
> web_switch_company_warning/static/src/js/switch_company_warning_worker.js:3: [W7903(javascript-lint), ] Expected space or tab after '//' in comment. [Error/spaced-comment]
> web_switch_company_warning/static/src/js/switch_company_warning_worker.js:5: [W7903(javascript-lint), ] Implicit global variable, assign as global property instead. [Error/no-implicit-globals]
> web_switch_company_warning/static/src/js/switch_company_warning_worker.js:6: [W7903(javascript-lint), ] Implicit global variable, assign as global property instead. [Error/no-implicit-globals]
> web_switch_company_warning/static/src/js/switch_company_warning_worker.js:8: [W7903(javascript-lint), ] Use the function form of 'use strict'. [Error/strict]
> web_switch_company_warning/static/src/js/switch_company_warning_worker.js:8: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
> web_switch_company_warning/static/src/js/switch_company_warning_worker.js:17: [W7903(javascript-lint), ] Line 17 exceeds the maximum line length of 80. [Error/max-len]

</details>

**Review by LoisRForgeFlow — APPROVED** [review]
:+1:

**Comment by LoisRForgeFlow** [comment]
@tbaden is it good to go for you?

**Comment by tbaden** [comment]
just a view warnings left:

<details><summary>logs</summary>

```

************* Module web_switch_context_warning
web_switch_context_warning/static/src/js/switch_context_warning.js:18: [W7903(javascript-lint), ] Line 18 exceeds the maximum line length of 80. [Error/max-len]
web_switch_context_warning/static/src/js/switch_context_warning.js:33: [W7903(javascript-lint), ] Line 33 exceeds the maximum line length of 80. [Error/max-len]
web_switch_context_warning/static/src/js/switch_context_warning.js:38: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]
web_switch_context_warning/static/src/js/switch_context_warning.js:46: [W7903(javascript-lint), ] Line 46 exceeds the maximum line length of 80. [Error/max-len]
web_switch_context_warning/static/src/js/switch_context_warning.js:47: [W7903(javascript-lint), ] Unexpected if as the only statement in an else block. [Error/no-lonely-if]
web_switch_context_warning/static/src/js/switch_context_warning.js:53: [W7903(javascript-lint), ] Missing trailing comma. [Error/comma-dangle]
web_switch_context_warning/static/src/js/switch_context_warning_worker.js:3: [W7903(javascript-lint), ] Implicit global variable, assign as global property instead. [Error/no-implicit-globals]
web_switch_context_warning/static/src/js/switch_context_warning_worker.js:4: [W7903(javascript-lint), ] Implicit global variable, assign as global property instead. [Error/no-implicit-globals]
web_switch_context_warning/static/src/js/switch_context_warning_worker.js:20: [W7903(javascript-lint), ] Missing trailing comma. [Error/comma-dangle]

```
</details>

not mandatory at the moment, but could be in the future. up to you if you want to fix these beforehand.

File: web_switch_context_warning/static/src/xml/switch_company_warning.xml [hunk 2]
```diff
@@ -1,8 +1,8 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (daramousk) [inline web_switch_context_warning/static/src/xml/switch_company_warning.xml:1]: This file has a different name than the one declared in the manifest, probably this is why I am getting a:

`Uncaught Error: QWeb2: Template 'web_switch_context_warning.warningWidget' not found`

File: web_switch_context_warning/static/src/js/switch_context_warning_worker.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+// Show a big banner in the top of the page if the company has been
+// changed in another tab or window (in the same browser)
+var con = [];
+var lastCtx = null;
+
+addEventListener("connect", function (ee) {
```
Comment (daramousk) [inline web_switch_context_warning/static/src/js/switch_context_warning_worker.js:6]: Who fires this `connect` event? This does not get executed for me (in a vanilla Odoo instance without anything else installed). That stops me from testing further.

**Pull request closed without merging**